### PR TITLE
bpo-47217: add name attribute to bz2 fileobject

### PR DESCRIPTION
**Pull request by ellaellela**
https://bugs.python.org/issue47217

### CODE REVIEW

File: Lib/bz2.py [hunk 1]
```diff
@@ -94,6 +94,11 @@ def __init__(self, filename, mode="r", *, compresslevel=9):
         else:
             self._pos = 0
 
+        if not isinstance(filename, (str, bytes)):
+            self.name = ''
```
Comment (merwok) [inline Lib/bz2.py:98]: Could this be some recognizable sentinel (`<stream>` or something better) rather than empty string?

Comment (ellaellela) [reply Lib/bz2.py:98]: This seemed the easiest way of doing it without complicating the logic and the tests. Also, I couldn't find a uniform way of dealing with a lack of "name" for `BytesIO()` objects in other libraries, so I did not want to introduce anything "new and specific".
Do you have an example (or an idea) of a sentinel use that would fit here? In that case, we could do a similar change for `gzip`, `bz2`, `lzma`, ... libraries to have a bit more standardized way of dealing with this situation when we have `BytesIO()` objects.

**Review by MaxwellDupre — APPROVED** [review]
Ran 98 tests in 11.650s
OK
Looks ok.

**Comment by serhiy-storchaka** [comment]
Thank you for your PR @ellaellela, but this is a part of more complex issue #115961. It is better to solve it all in one place.

**Pull request closed without merging**